storage/client/queue/src/main/java/com/azure/storage/common/credentials/SharedKeyCredential.java

```diff
@@ -111,8 +113,10 @@ public String computeHmac256(final String stringToSign) throws InvalidKeyExcepti
             hmacSha256.init(new SecretKeySpec(this.accountKey, "HmacSHA256"));
             byte[] utf8Bytes = stringToSign.getBytes(StandardCharsets.UTF_8);
             return Base64.getEncoder().encodeToString(hmacSha256.doFinal(utf8Bytes));
-        } catch (final  NoSuchAlgorithmException e) {
-            throw new Error(e);
+        } catch (final NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        } catch (InvalidKeyException e) {
+            throw new RuntimeException("Please double check the account key. Error details: " + e.getMessage());
```

Comment:
Use the new `ClientLogger.logAndThrow` API whenever you throw an exception.

Reply:
Done.